waku/waku_test.go

```diff
@@ -1065,6 +1068,31 @@ func TestTopicInterest(t *testing.T) {
 
 }
 
+func TestOnNewEnvelopesSoftBlacklist(t *testing.T) {
+	w1 := New(nil, nil)
+
+	envelope := &common.Envelope{}
+	p2pPeer := p2p.NewPeer(enode.ID{0x4}, "test", []p2p.Cap{})
+	peer := v1.NewPeer(w1, p2pPeer, nil, nil)
+
+	// Pre-condition, we need to make sure this envelope returns an EnvelopeError
+	envelopeError, err := w1.OnNewEnvelopes([]*common.Envelope{envelope}, peer)
+	require.NoError(t, err)
+	// Make sure this envelope returns an error
+	require.NotNil(t, envelopeError)
+
+	// build black listed waku
+	cfg := &Config{
+		SoftBlacklistedPeerIDs: []string{types.EncodeHex(peer.ID())},
+	}
+	w2 := New(cfg, nil)
+
+	envelopeError, err = w2.OnNewEnvelopes([]*common.Envelope{envelope}, peer)
+	require.NoError(t, err)
+	// Since it's blacklisted, it will just drop envelopes, keep the connection open
+	require.Nil(t, envelopeError)
+}
```

Comment:
Would it make sense to test that the peers are still connected?

Reply:
Connection is handled at the peer level (so v0/v1), in this case as long as we don't return `err` from `OnNewEnvelopes` the peer will not be disconnected, so we should be safe as implicitly tested